### PR TITLE
Validator page responsiveness

### PR DESCRIPTION
**Pull request by eshark9312**
- #4595

[> Design](https://www.figma.com/file/QnjZuWYtGimTEkQlN6MNlH/Validators?type=design&node-id=5-1550&mode=design&t=J4APDC39U67GiH4c-0)
[> Preview](https://dao-git-fork-eshark9312-validator-page-responsive-joystream.vercel.app/#/validators)

- Added validator page to the sidebar menu.
- Added responsiveness.

### CODE REVIEW

File: packages/ui/src/validators/modals/ValidatorsInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+              <TextMedium>
+                When nominating, you are at risk of having parts of your staked funds lost if the validator malfunctions
+                or does a poor job, resulting in a reduced return on investment. To manage your risk, we advice you to
+                nominate several validators (up to 16). This allows you to spread out your risk and increase your
+                chances of earning rewards. You can choose how much to stake with each validator, and you can change
+                your staking percentages at any time.
+              </TextMedium>
+              <TextMedium>
+                To begin, review each validator's performance metrics by clicking on their name in the list. When you're
+                ready to nominate, add the validators you'd like to nominate to by clicking the "Nominate" button on the
```
Comment (thesan) [inline packages/ui/src/validators/modals/ValidatorsInfo.tsx:50]:
```suggestion
                ready to nominate, add the validators you'd like to nominate by clicking the "Nominate" button on the
```

File: packages/ui/src/validators/modals/ValidatorsInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                list or directly on the validator’s profile. Once you've selected your validators, click the "Proceed"
```
Comment (thesan) [inline packages/ui/src/validators/modals/ValidatorsInfo.tsx:51]:
```suggestion
                list or directly on the validator’s profile. Once you've selected a validator, click the "Proceed"
```

File: packages/ui/src/validators/modals/ValidatorsInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react'
+import { Link } from 'react-router-dom'
```
Comment (thesan) [inline packages/ui/src/validators/modals/ValidatorsInfo.tsx:2]:
```suggestion
import { Link } from '@/common/components/Link'
```

File: packages/ui/src/validators/modals/ValidatorsInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+              <TextMedium>
+                To begin, review each validator's performance metrics by clicking on their name in the list. When you're
+                ready to nominate, add the validators you'd like to nominate to by clicking the "Nominate" button on the
+                list or directly on the validator’s profile. Once you've selected your validators, click the "Proceed"
+                button to initiate the nomination process.
+              </TextMedium>
+            </RowGapBlock>
+            <TextMedium>
+              You can learn more about the Pioneer nomination{' '}
+              <Link to="#" style={{ color: Colors.Blue[500], textDecoration: 'underline' }}>
```
Comment (thesan) [inline packages/ui/src/validators/modals/ValidatorsInfo.tsx:57]:
```suggestion
              <Link href="https://handbook.joystream.org/system/nomination">
```

**Review by thesan — CHANGES_REQUESTED** [review]
The columns are really crammed together:
image
There should be a gap on this grid.
And there should also be a gap after "Verification (?)":
image

Also I noticed that it takes a while to get the validators. The list is loading for 20+ seconds. I'm assuming the rpc node is why the query is slow, but when I switch page the page loads again even though the provider should have the data already... Could you check what's going wrong please ?

**Comment by eshark9312** [comment]
@thesan 
The first loading takes 20+ seconds and the next time takes about 10s. That's due to `useValidatorsList` hook which fetches the detailed information for all validators. The `provider` only stores the validator addresses and memberships.
Do you want all data to be stored in the `provider` instead of `useValidatorsList` hook?

**Comment by thesan** [comment]
> Do you want all data to be stored in the provider instead of useValidatorsList hook?

Yes I think it would be good. Just make sure the add validator modal doesn't fetch the extra data please.

**Review by thesan — APPROVED** [review]
Sorry for the delay on this PR I thought you were still working on it.

It looks good. I'll create a PR to speed up the page render and then the QA can start hopefully next week :crossed_fingers:.

This dashboard was tricky and the work wasn't well defined from the start but you did a great job. Thanks a lot :pray:

**Pull request merged**